src/sagemaker_containers/_mpi.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import argparse
+import inspect
+import logging
+import os
+import subprocess
+import time
+from typing import Any, List, Tuple  # noqa ignore=F401 imported but unused
+
+import paramiko
+import psutil
+
+import libchangehostname
+from sagemaker_containers import _logging, _process, _timeout
+
+logger = _logging.get_logger()
+logging.getLogger("paramiko").setLevel(logging.INFO)
+
+
+class WorkerRunner(_process.ProcessRunner):
+    """Runner responsible prepare MPI training and wait for MPI master execution
+    """
+
+    def __init__(self, user_entry_point, args, env_vars, master_hostname):
+        super(WorkerRunner, self).__init__(user_entry_point, args, env_vars)
+        self._master_hostname = str(master_hostname)
+
+    def run(self, wait=True, capture_error=False):  # type: (bool, bool) -> None
+        """The WorkerRunner proceeds as following:
+
+        - wait for the MPI Master to create its SSH daemon
+        - start its SSH daemon
```

Comment:
I'd reorder and reword these:

- Start the worker's SSH daemon
- Wait for the MPI Master to create its own SSH daemon.
...